pkg/backend/httpstate/state.go

```diff
@@ -202,11 +202,23 @@ func (u *cloudUpdate) RecordAndDisplayEvents(
 			}
 			displayEvents <- diagEvent
 		}
+	}
+
+	for e := range events {
+		// First echo the event to the local display.
+		displayEvents <- e
+
+		// Then render and record the event for posterity.
+		eventIdx++
+		go recordEngineEvent(e, eventIdx)
```

Comment:
Doing this asynchronously means that we're now potentially writing to `displayEvents` at some arbitrary time in the future after receiving the event - we've had a bunch of crashes in the past because components fail to shut down gracefully and try to write to `displayEvents` when the CLI is shutting down. Is t here a way we can gracefully shut down the spawned goroutine if we know that the CLI is in the process of shutting down?

Reply:
I don't think we can do that without passing additional state, e.g. a `context.Context` object. 

What would you think of either (1) silently dropping these errors -- meaning the stack's update logs might have "gaps" in the timeilne view -- or (2) just printing the error to `STDERR` (and potentially corrupting the display events channel).

Ultimately if we don't trust the `displayEvents` channel to be open or not, we probably can't reliably use it in these separate Go-routines.

Reply:
> What would you think of either (1) silently dropping these errors -- meaning the stack's update logs might have "gaps" in the timeilne view -- or (2) just printing the error to STDERR (and potentially corrupting the display events channel).

I think (1) makes the most sense here IMO - that sounds better to me than garbling up standard error. Given that this is ostensibly usable as a library I think it makes sense to avoid writing out to standard error if possible.

Reply:
🥇